### ISSUE TITLE
Rotary games

### ISSUE THREAD

**Issue by arcadez** [libretro/mame2003-plus-libretro#47]
These can be a pain as the controls dont transfer to well to gamepads via the MAME inputs
anyway i have some code handy which for Downtown and Ikari Warriors III would just be 
dropping into the source a couple of new Roms which are official versions of these games
but with joystick controls rather than rotary.

In the case of the following these would be Jap bootlegs of some famous SNK games which
hack out the rotary controls replacing them with again a normal joystick control setup
apart from the new roms this would require me to backport some new input code for the
games into the snk.c driver but it's doable 

Ikari Warriors
Guerilla War
Victory Road

These are all supported in FBA already but should you fancy them for this ole core also
then lemmie know and i'll add em shouldn't take more than a few hours to port the code
and test the games.

**Comment by markwkidd** [comment]
There are people who use mame2003 with actual rotary dials.

No problem maybe: Can the new code be a libretro core option? Default is
joystick, core option changes these games to original rotary mode?

**Comment by dankcushions** [comment]
even better: just accept both inputs. if they use _left/right, it works, if they use _anolog_left/right, it works, and if they use mouse it works. shouldn't need to choose.

**Comment by arcadez** [comment]
Oh im not the best at explaining things sometimes or maybe all the time :) with regards to
Downtown and Ikari Warriors III it's just a rom drop into the driver as they hang off the same
code as the rotary versions of those games

The SNK bootlegs however would need inputs and joystick code backported into the driver
but again this doesn't actually affect any of the rotary games which are already supported as
the new input code would only be used by the bootlegs.

Basically in a nutshell there will be no changes to the corewide inputs or the existing rotary
games all this means is you can play say for example Ikari Warriors as if you were playing
Capcom's Commano eg you move and turn via the 8-way joystick with A for fire and B for Bomb

**Comment by Wilstorm** [comment]
I do fancy them, some true arcade classics!

**Comment by ZappaUtopia** [comment]
All Joystick versions would be welcome !

I just tried the Joystick Bootleg of Guerilla War and it seems to control the same as the rotary version, it only shoots straight ahead and you have to map a button to rotate left/right. So I don't think it works properly.

**Comment by arcadez** [comment]
Ok i'll be making the commit shortly all these will now be supported.........

Battle Shark (Japan, Joystick)
Dogou Souken (Joystick, Hack Bootleg)  Dogou Souken is Victory Road in japan
Downtown (Joystick, Hack)
Guerilla War (Joystick, Hack Bootleg)
Ikari Warriors (Joystick, Hack Bootleg)
Ikari Warriors III (World, 8-Way Joystick)

No core input or rotary code has been touched, the new rotate handling will only be used
by the above games were applicable and none of the currantly supported rotary games will
be affected in any way shape or form.

Finally no roms have changed over and above the new ones im going to add support for
although the description has changed for two of them this bring them inline with currant
MAME to reflect the fact that they are indeed joystick bootlegs.

GAMEX( 1987, gwarb,    gwar,     gwar,     gwar,     gwar,     ROT270, "bootleg", "Guerrilla War (bootleg)", GAME_NO_COCKTAIL ) 

Becomes

GAMEX( 1987, gwarb,    gwar,     gwar,     gwarb,    gwarb,    ROT270, "bootleg", "Guerrilla War (joystick hack bootleg)", GAME_NO_COCKTAIL )

GAMEX( 1986, ikarijpb, ikari,    ikari,    ikarijp,  ikarijpb, ROT270, "bootleg", "Ikari (Japan bootleg)", GAME_NO_COCKTAIL )

Becomes

GAMEX( 1986, ikarijpb, ikari,    ikari,    ikarijp,  ikarijpb, ROT270, "bootleg", "Ikari (joystick hack bootleg)", GAME_NO_COCKTAIL )

**Comment by markwkidd** [comment]
Yes please! There are plenty of games that are part of mame2003 already
with worse bugs than one which makes it hard to complete level 3 :)

**Comment by grant2258** [comment]
that was quick work mr arcadze!

**Comment by arcadez** [comment]
I have all the code in another project bigman, all i have to do is rework some of it for the
MAME78 codebase then test the games before porting the new stuff across ;).

**Comment by grant2258** [comment]
still is damn fast!

**Comment by arcadez** [comment]
Aye i thought i'd better get the code in now incase Trump starts WWIII :)

**Comment by arcadez** [comment]
@markwkidd no problem i'll add it in later today

**Comment by grant2258** [comment]
@arcadez  have to admit trumps going to be a legend for the wrong reasons lol

**Comment by arcadez** [comment]
Alright i've added the Midnight Resistance joystick bootleg, i forgot just how nasty it is expect
some graphical problems here and there as they hacked out a full graphical layer, the music
is different also plus kinda buggy you'll notice it on the boss fights as pressing fire makes it stutter.

I guess the bootleg D team were assigned this one :)
